layers/+window-management/exwm/funcs.el

```diff
@@ -78,6 +79,63 @@ Can show completions at point for COMMAND using helm"
   (interactive)
   (start-process "" nil exwm-locking-command))
 
+(defun exwm/autostart-process (name command &optional directory)
+  "Run COMMAND as a process with NAME.
+
+The started process is also added to `exwm-autostart-process-list`.
+
+DIRECTORY can be set to a string which will be used as working directory for the
+process.  If not supplied, will be set to `user-home-directory'."
+  (push (let ((default-directory (or directory user-home-directory)))
+          (start-process-shell-command name nil command))
+        exwm-autostart-process-list))
+
+(defun exwm//start-desktop-application (basename xdg)
+  "Autostart an application from a XDG desktop entry specification."
+  (when-let* ((type (gethash "Type" xdg))
+              (_ (string= type "Application"))
+              (cmd (gethash "Exec" xdg))
+              (directory (gethash "Path" xdg))
+              ;; (name (gethash "Name" xdg))
+              ;; (hidden (gethash "Hidden" xdg))
+              ;; (dbus-p (gethash "DBusActivatable" xdg)) ; TODO: support
+              (let ((include (gethash "OnlyShowIn" xdg))
+                    (exclude (gethash "NotShowIn" xdg))
+                    (try-exec (gethash "TryExec" xdg)))
+                (and (cond (include (member "EXWM" (split-string include ";" t)))
+                           (exclude (not (member "EXWM" (split-string exclude ";" t))))
+                           (t))
+                     (if try-exec
+                         (executable-find try-exec)
+                       t))))
```

Comment:
I think these are conforming to the autostart specification:
https://specifications.freedesktop.org/autostart-spec/autostart-spec-latest.html

So you need to also check the "Hidden Key".
Basically, the logic is
- If "Hidden", end early
- If "OnlyShowIn", but doesn't match, end early
- If "NotShowIn", and matches, end early
- If "TryExec", but the program is not found, end early

Thus it's possible to add those into the `when-let*` form.
```elisp
(when-let* ((type (gethash "Type" xdg))
            (_application? (string= type "Application")
            (cmd (gethash "Exec" xdg))
            (directory (gethash "Path" xdg user-home-directory))
            (_include? (and (null (gethash "Hidden" xdg))
                            (if-let ((only-show (gethash "OnlyShowIn" xdg)))
                                (member "EXWM" (split-string only-show ";" t))
                              t)
                            (if-let ((not-show (gethash "NotShowIn" xdg)))
                                (not (member "EXWM" (split-string not-show ";" t)))
                              t)
                            (if-let ((try-exec (gethash "TryExec" xdg)))
                                (executable-find try-exec)
                              t))))
  (exwm/autostart-process basename cmd directory))
```

Check the spelling and parentheses. I typed them by hand.